src/server/api/modules/recipes/service/scraper.ts

```diff
@@ -10,13 +10,25 @@ import { JSDOM } from "jsdom"
 
 import { logger } from "~/lib/logger"
 
+function normalizeWhitespace(input: string): string {
+  return input.replace(/[\u00A0\u1680\u180E\u2000-\u200A\u2028\u2029\u202F\u205F\u3000]/g, ' ');
+}
+
+function escapeRealLineBreaksInString(input: string): string {
+  return input.replace(/(["'])([\s\S]*?)\1/g, (match, quote, content) => {
+    const escapedContent = content.replace(/\n/g, "\\n").replace(/\r/g, "\\r");
+    return quote + escapedContent + quote;
+  });
+}
+
```

Comment:
this code made it in when #503 was merged - rebasing on the new master should hopefully remove it from this diff